src/lib/comms/sol-http-server-impl-microhttpd.c

```diff
@@ -959,6 +1058,81 @@ sol_http_server_send_response(struct sol_http_request *request, struct sol_http_
     return ret;
 }
 
+SOL_API struct sol_http_response_sse *
+sol_http_server_send_response_sse(struct sol_http_request *request,
+    void (*cb)(void *data, const struct sol_http_response_sse *sse),
+    const void *cb_data)
+{
+    int ret;
+    struct sol_http_response_sse *sse;
+    struct MHD_Response *mhd_response;
+
+    SOL_NULL_CHECK(request, NULL);
+    SOL_NULL_CHECK(request->connection, NULL);
+
+    sse = malloc(sizeof(*sse));
+    SOL_NULL_CHECK(sse, NULL);
+
+    sol_buffer_init(&sse->buffer);
+    sse->connection = request->connection;
+    sse->delete_me = false;
+
+    MHD_resume_connection(request->connection);
+
+    mhd_response = build_mhd_response_sse(sse);
+    SOL_NULL_CHECK_GOTO(mhd_response, err);
+
+    ret = MHD_queue_response(request->connection, SOL_HTTP_STATUS_OK, mhd_response);
+    MHD_destroy_response(mhd_response);
+
+    SOL_INT_CHECK_GOTO(ret, != MHD_YES, err);
+
+    sse->cb = cb;
+    sse->cb_data = cb_data;
+
+    return sse;
+
+err:
+    free(sse);
+    return NULL;
+}
+
+SOL_API int
+sol_http_response_sse_del(struct sol_http_response_sse *sse)
+{
+    SOL_NULL_CHECK(sse, -EINVAL);
```

Comment:
maybe also check `delete_me` to avoid "double-free"? It won't fail and hint at valgrind as the memory may still be alive, so if we can detect the `delete_me` was already set, we can report to user.